### PR TITLE
fix: convert email address to lowercase before any Cognito API request

### PR DESCRIPTION
**Pull request by craigzour**
# Summary | Résumé

closes https://github.com/cds-snc/platform-forms-client/issues/2269

- Added email address sanitization when making Cognito API requests
- Added Prisma migration function to modify existing email addresses from uppercase to lowercase (will be auto executed when we deploy in production)
- Added Cognito utility script to modify existing email addresses from uppercase to lowercase (will need to be manually ran when we push this fix to production)

FYI: Both migrations have been tested locally.

# Test instructions | Instructions pour tester la modification

- Try to sign in with your regular account but add random uppercase letters in your email address. Sign in should be successful
- Register with an email address that contains uppercase letters. Check Cognito using the AWS UI and you should see your account with the email address converted to lowercase

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [ ] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [ ] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

**Comment by craigzour** [comment]
To remember: when we push the fix to production we will have to run the Cognito migration script.
cc @bryan-robitaille @timarney @thiessenp-cds

File: utils/cognitoMigration/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
```
Comment (timarney) [inline utils/cognitoMigration/tsconfig.json:3]: Do we want to keep all the commented out lines?

Comment (craigzour) [reply utils/cognitoMigration/tsconfig.json:3]: I just used an existing template but I clean it up since it won't be necessary

**Comment by timarney** [comment]
Nice work on the solve -- will wait for others to have a look.

**Review by timarney — APPROVED** [review]

**Comment by craigzour** [comment]
@timarney will let Bryan do a final review (as discussed with him) and merge when he is done.

**Review by thiessenp-cds — APPROVED** [review]
LGTM :)
Nice work, especially the migration logic!

**Review by bryan-robitaille — APPROVED** [review]
Tested locally, works great.
Modified readme slightly to remove bullet point that was referencing a different utility script.

**Pull request merged**